File: java/basics/LargestTwoNumbers01.java
package basics;

import java.util.Scanner;

public class LargestTwoNumbers01 {

	public static void main(String[] args) {
		System.out.println("Enter two numbers:");
		Scanner scn = new Scanner(System.in);
		int num1 = scn.nextInt();
		int num2 = scn.nextInt();
		scn.close();
		
		System.out.println(findLargest01(num1, num2));
		System.out.println(findLargest02(num1, num2));
		System.out.println(findLargest03(num1, num2));
	}

	private static int findLargest01(int num1, int num2) {
		// this if is in separate line to cover this we can write >= instead
		if (num1 == num2) { 
			return num1;
		} else if (num1 > num2) {
			return num1;
		} else {
			return num2;
		}
	}
	
	private static int findLargest02(int num1, int num2) {
		// this if is in separate line to cover this we can write >= instead
		if (num1 == num2) { 
			return num1;
		}
		
		int res = num1 > num2?num1:num2;
		return res;
	}
	
	private static int findLargest03(int num1, int num2) {
		// this if is in separate line to cover this we can write >= instead
		if (num1 == num2) { 
			return num1;
		}
		
		int res = Math.max(num1, num2);
		return res;
	}

}

File: python/LargestTwoNumbers01.py
print('Enter the two numbers')
num1 = int(input())
num2 = int(input())


def findLargest01(num1, num2):
    largest = -1
    
    if num1 >= num2:
        largest = num1
    else:
        largest = num2
        
    return largest   

def findLargest02(num1, num2):
    return num1 if num1 >= num2 else num2

def findLargest03(num1, num2):
    return max(num1, num2)
    
def findLargest04(num1, num2):
    largestNum = lambda a, b:a if a >= b else b
    return largestNum(num1, num2)

# list comprehension
def findLargest05(num1, num2):
    x = [num1 if num1 >= num2 else num2]
    return x[0]

def findLargest06(num1, num2):
    x = [num1, num2]
    x.sort()
    return x[-1]
    
        

print(findLargest01(num1, num2))
print(findLargest02(num1, num2))
print(findLargest03(num1, num2))
print(findLargest04(num1, num2))
print(findLargest05(num1, num2))
print(findLargest06(num1, num2))
        
    
